src/devices/vdml_usd.cpp

```diff
@@ -21,5 +21,80 @@ std::int32_t is_installed(void) {
 	return usd_is_installed();
 }
 
+std::int32_t list_files_raw(const char* path, char* buffer, int32_t len) {
+	return usd_list_files_raw(path, buffer, len);
+}
+
+std::vector<std::string> list_files(const char* path, char* buffer, int32_t len) {
+	std::vector<std::string> files = {};
+	// Call the C function
+	int32_t success = usd_list_files_raw(path, buffer, len);
+	// Check if call successful, if error return vector containing error state
+	if (success == PROS_ERR) {
+		// Check errno to see which error state occurred
+		// push back error state to files vector as std::string
+		switch (errno) {
+			case EIO:
+				files.push_back("EIO");
+				break;
+			case EINVAL:
+				files.push_back("EINVAL");
+				break;
+			case EBUSY:
+				files.push_back("EBUSY");
+				break;
+			case ENOENT:
+				files.push_back("ENOENT");
+				break;
+			case EACCES:
+				files.push_back("EACCES");
+				break;
+			case EEXIST:
+				files.push_back("EEXIST");
+				break;
+			case EROFS:
+				files.push_back("EROFS");
+				break;
+			case ENXIO:
+				files.push_back("ENXIO");
+				break;
+			case ENOBUFS:
+				files.push_back("ENOBUFS");
+				break;
+			case ENFILE:
+				files.push_back("ENFILE");
+				break;
+			default:
+				// If none of the above, will be FILE IO error
+				files.push_back("FILE I/O ERROR");
+				break;
+		}
+		return files;
+	}
+	// Parse buffer given call successful, split by '/n'
+	// Store char * buffer as std::string in str
+	// Store delimiter in std::string, '\n'
+	std::string str(buffer);
+	std::string delimiter = "\n";
+
+	// position to keep track of position in str
+	// file_name used to store each file name
+	size_t position = 0;
+	std::string file_name;
+
+	// Loop until delimiter '\n' can not be found anymore
+	while ((position = str.find(delimiter)) != std::string::npos) {
+		// file_name is the string from the beginning of str to the first '\n', excluding '\n'
+		file_name = str.substr(0, position);
+		// Add token to files vector
+		files.push_back(file_name);
+		// Erase file_name from str, would be pos + 1 if we wanted to include '\n'
+		str.erase(0, position + 1);
```

Comment:
This won't work if you use a `std::string_view` for str (and it's very unnecessarily slow even as a `std::string`). You should either keep track of the index to start at for the next iteration, or maybe `str = str.substr(...)`?

Reply:
Implemented this change in commit 3d41ca62de29da852509b02fcd8ce85c94a92e60. Altered code to instead have index to keep track of start for next iteration.

Reply:
Very nice! Thanks